backend/uclapi/timetable/models.py

```diff
@@ -0,0 +1,555 @@
+
+
+class Stumodules(models.Model):
+    setid = models.TextField(max_length=10)
+    studentid = models.TextField(primary_key=True, max_length=12)
+    deptid = models.TextField(max_length=10)
+    moduleid = models.TextField(max_length=12)
+    modgrpcode = models.TextField(max_length=10)
+    slotid = models.BigIntegerField()
+    fixingrp = models.CharField(max_length=1)
+    modpart = models.TextField(max_length=10)
+    restype = models.TextField(max_length=10)
+    unitvalue = models.TextField(max_length=10)
+    classif = models.TextField(max_length=10)
+    papernum = models.BigIntegerField()
+    modlevel = models.TextField(max_length=10)
+    inactive = models.CharField(max_length=1)
+    instid = models.BigIntegerField()
+    courseid = models.TextField(max_length=12)
+    crsyear = models.BigIntegerField()
+    semid = models.BigIntegerField()
+    moddropped = models.CharField(max_length=1)
+    donotcount = models.CharField(max_length=1)
+    semrank = models.BigIntegerField()
+
+    class Meta:
+        managed = False
+        db_table = 'NOSH_CMIS_STUMODULES'
+        _DATABASE = 'roombookings'
+
+
+class Features(models.Model):
+    setid = models.TextField(max_length=10)
+    featureid = models.TextField(max_length=10)
+    description = models.TextField(primary_key=True, max_length=80)
+    linkcode = models.TextField(max_length=20)
+    contactid = models.TextField(max_length=10)
+    cost = models.TextField(max_length=10)
+
+    class Meta:
+        managed = False
+        db_table = 'NOSH_CMIS_FEATURES'
+        _DATABASE = 'roombookings'
+
+
+class Equipment(models.Model):
+    setid = models.TextField(max_length=10)
+    equipid = models.TextField(primary_key=True, max_length=10)
+    description = models.TextField(max_length=80)
+    linkcode = models.TextField(max_length=20)
+    contactid = models.TextField(max_length=10)
+    cost = models.TextField(max_length=10)
+
+    class Meta:
+        managed = False
+        db_table = 'NOSH_CMIS_EQUIPMENT'
+        _DATABASE = 'roombookings'
+
+
+class Course(models.Model):
+    setid = models.TextField(max_length=10)
+    courseid = models.TextField(primary_key=True, max_length=12)
+    name = models.TextField(max_length=80)
+    category = models.TextField(max_length=10)
+    type = models.TextField(max_length=10)
+    linkcode = models.TextField(max_length=20)
+    owner = models.TextField(max_length=10)
+    numyears = models.BigIntegerField()
+    crsweight = models.FloatField()
+    minmodules = models.BigIntegerField()
+    maxmodules = models.BigIntegerField()
+    numplaces = models.BigIntegerField()
+    mintotal = models.BigIntegerField()
+    maxtotal = models.BigIntegerField()
+    firstyear = models.BigIntegerField()
+    oldcourseid = models.TextField(max_length=12)
+    isactive = models.CharField(max_length=1)
+    lecturerid = models.TextField(max_length=10)
+
+    class Meta:
+        managed = False
+        db_table = 'NOSH_CMIS_COURSE'
```

Comment:
can we move back to `'"CMIS_OWNER"."COURSE"'` and the like?

Reply:
let me try that now

Reply:
done. but not helping in speed haha